src/transform.nit

```diff
@@ -285,6 +285,57 @@ redef class AForExpr
 	end
 end
 
+redef class AWithExpr
+	# is replaced with a do/end and injected calls to `start` and `finish`
+	#
+	# Basically, the following
+	#
+	# ~~~nitish
+	# with expr do
+	#   block
+	# end label l
+	# ~~~
+	#
+	# is transformed into
+	#
+	# ~~~nitish
+	# var x = expr
+	# do
+	#   x.start
+	#   block
+	# end label l
+	# x.finish
+	# ~~~
+	#
+	# The point is that `finish` is called even if the block is escaped.
```

Comment:
Maybe you wanted a way that the provider of the finishable can skip the block in some condition and document it this way.

eg
~~~
# Force the enclosed of code to be executed while playing `Ride of the Valkyries` at max sound volume
# If there is no sound card or it is busy or cannot be set to max volume, then the whole sequence of code is not executed because the user *does not deserve it*.
fun grandiose: Finishable do ...
~~~

so the client can just write

~~~
with grandiose do
   # some important computation like cyptoanalysis
   ...
end
~~~

But this cause issue with how to do it, and more important to me, how making it usable with the expected local control-flow. I want that the semantic of the program matches the expectation of the programmer. For instance

~~~
var x
with something do
   x = 1
end
# is x initialized or not?
~~~

By saying that the `with` block is always entered once, I can have a POLA semantic on the control flow.